soc/riscv/riscv-privilege/telink_b9x/soc.c

```diff
@@ -109,6 +116,41 @@ void soc_load_rf_parameters_deep_retention(void)
 }
 #endif
 
+bool flash_cap_check(void) {
+	bool status  = false;
+	uint32_t data = flash_read_mid();
+	uint8_t flash_cap = (uint8_t)(data >> FLASH_SIZE_OFFSET);
+	size_t bin_size = FIXED_PARTITION_SIZE(flash);
```

Comment:
more clean is to use chosen
zephyr,flash

Reply:
Thank you, changed to DT_REG_SIZE(DT_CHOSEN(zephyr_flash)